enhancements/olm-toggle-copied-csvs.md

```diff
@@ -0,0 +1,163 @@
+- Docs suggest admins notify their cluster tenants of installed operators directly; i.e. email, slack, etc
+
+#### OpenShift Console Integration
+
+**Risk:** The OpenShift Console uses Copied CSVs in the rendering of certain pages.
+**Mitigation:** Consult Console maintainers prior to deprecating Copied CSVs.
+
+## Drawbacks
+
+- [Loss of Operator Discovery](#loss-of-operator-discovery)
+- Relies on the introduction of a new `ConfigMap` for configuring OLM
+
+## Alternatives
+
+### Command-line Option Only
+
+Update the olm-operator `Deployment` to toggle the Copied CSV feature.
+
+#### Cons
+
+On an OpenShift cluster, [a CVO override](https://docs.openshift.com/container-platform/4.8/logging/config/cluster-logging-maintenance-support.html#unmanaged-operators_cluster-logging-unsupported) would be required to prevent the updated command-line option in the olm-operator `Deployment` from reverting, but this also puts clusters into an unsupported state and prevents cluster upgrades.
+
+### OpenShift FeatureGates
+
+Define a new OpenShift [FeatureGate](https://github.com/openshift/api/blob/master/config/v1/types_feature.go) feature set that, when disabled, disables Copied CSVs. This would be preferred over any command-line option and would require no CVO override.
+
+#### Cons
+
+- Requires OLM to be OpenShift-aware
+  - Could be addressed with a downstream-only controller that deploys OLM with the correct options
+- Requires changes OpenShift (a new feature set)
+
+### OLM Config CRD
+
+Add a new CRD to define OLM-specific configuration instead of using the `olm-config` `ConfigMap`, adding more type-safety.
+
+#### Cons
+
+- Adds a new API to maintain
+- Adding APIs is questionably backportable (for OpenShift releases)
+- Singleton CRs beg for singleton enforcement, which today requires validating webhooks, increasing implementation complexity
+
+### New Field (or Annotation)
+
+Add a new field or annotation to existing types such as `CSVs` or `OperatorGroups` to denote disabling/enabling Copied CSVs for related resources.
+
+#### Cons
+
+- CVO would revert changes to resources included with the payload; e.g. the `openshift-operators` `Namespace` or `OperatorGroup`
+- Changing existing schemas is questionably backportable (for OpenShift releases)
+- Giving the ability to toggle the feature for some portions of a cluster but not others opens the door to an inconsistent user experience; e.g. as a user, I must use two separate methods to be sure of operator availablity.
+
+### Virtual "InstalledOperator" API
+
+A "virtual resource" served by an aggregated API -- much like `PackageManifest` -- that provides a namespaced, CSV-like, view of the operators configured to interact with a given namespace. Such a resource would be a direct drop-in-place for a Copied CSV.
+
+#### Cons
+
+- Aggregated APIs generally lead to cluster-instability
+- Aggregated APIs require serving cert management and rotation; i.e. increased installation and lifecycle burden
```

Comment:
I think we could enable users to opt-out of this feature in such a case, which would only put the burden on those users, which require that feature. Many platforms also come with on-board certificate management and rotation.

Reply:
Fair points; although, I was under the assumption that we've already discussed and dismissed this approach in other venues. Is that not the case?

Reply:
it feels like a big hammer of a solution to use in the short term, given the longer term direction we are taking in solving this problem.